core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -206,11 +206,12 @@ private[spark] class MemoryStore(
       }
       val size = entry.size
       def transferUnrollToStorage(amount: Long): Unit = {
```

Comment:
I'm actually going to just revert this change since it's fine to synchronize this transfer since the transfer is incapable of spilling.